pkg/controller/v1alpha1/localmodelnode/controller.go

```diff
@@ -192,60 +162,127 @@ func (c *LocalModelNodeReconciler) getContainerSpecForStorageUri(ctx context.Con
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	c.Log.Info("Found jobs", "model", modelName, "num of jobs", len(jobList.Items))
+	var latestJob *batchv1.Job
+	for i, job := range jobList.Items {
+		if excludeSucceeded && job.Status.Succeeded > 0 {
+			continue
+		}
+		if latestJob == nil || job.CreationTimestamp.After(latestJob.CreationTimestamp.Time) {
+			latestJob = &jobList.Items[i]
+		}
+	}
+	return latestJob, nil
+}
+
+func getModelStatusFromJobStatus(jobStatus batchv1.JobStatus) v1alpha1api.ModelStatus {
+	switch {
+	case jobStatus.Succeeded > 0:
+		return v1alpha1api.ModelDownloaded
+	case jobStatus.Failed > 0:
+		return v1alpha1api.ModelDownloadError
+	case jobStatus.Ready != nil && *jobStatus.Ready > 0:
+		return v1alpha1api.ModelDownloading
+	default:
+		return v1alpha1api.ModelDownloadPending
+	}
+}
+
 // Create jobs to download models if the model is not present locally
 // Update the status of the LocalModelNode CR
 func (c *LocalModelNodeReconciler) downloadModels(ctx context.Context, localModelNode *v1alpha1api.LocalModelNode) error {
 	newStatus := map[string]v1alpha1api.ModelStatus{}
 	for _, modelInfo := range localModelNode.Spec.LocalModels {
-		if status, ok := localModelNode.Status.ModelStatus[modelInfo.ModelName]; ok {
-			if status == v1alpha1api.ModelDownloaded {
-				newStatus[modelInfo.ModelName] = v1alpha1api.ModelDownloaded
-				continue
-			}
-		}
-
-		jobName := modelInfo.ModelName + "-" + localModelNode.ObjectMeta.Name
-
-		job, err := c.launchJob(ctx, jobName, localModelNode, modelInfo, modelInfo.ModelName)
+		var job *batchv1.Job
+		folderExists, err := fsHelper.hasModelFolder(modelInfo.ModelName)
 		if err != nil {
-			c.Log.Error(err, "Job error", "name", jobName)
+			c.Log.Error(err, "Failed to check model folder", "model", modelInfo.ModelName)
 			return err
 		}
-
-		switch {
-		case job.Status.Succeeded > 0:
-			newStatus[modelInfo.ModelName] = v1alpha1api.ModelDownloaded
-		case job.Status.Failed > 0:
-			newStatus[modelInfo.ModelName] = v1alpha1api.ModelDownloadError
-		case job.Status.Ready != nil && *job.Status.Ready > 0:
-			newStatus[modelInfo.ModelName] = v1alpha1api.ModelDownloading
-		default:
-			newStatus[modelInfo.ModelName] = v1alpha1api.ModelDownloadPending
+		if folderExists {
+			c.Log.Info("Model folder found", "model", modelInfo.ModelName)
+			// If folder exists and the job has been successfully completed, do nothing
+			if status, ok := localModelNode.Status.ModelStatus[modelInfo.ModelName]; ok {
+				if status == v1alpha1api.ModelDownloaded {
+					newStatus[modelInfo.ModelName] = v1alpha1api.ModelDownloaded
+				}
+				continue
+			}
+			job, err = c.getLatestJob(ctx, modelInfo.ModelName, nodeName, false)
+			if err != nil {
+				c.Log.Error(err, "Failed to getLatestJob", "model", modelInfo.ModelName, "node", nodeName)
+				return err
+			}
+			// If job is not found, create a new one. Because download could be incomplete.
```

Comment:
job can also be cleaned up by TTL, do we want to create download job in this case?

Reply:
Yes, job can also be cleaned up by TTL if the job was successful, we would have already updated the status of the model to be downloaded, which would bypass this piece of code. See the `continue` above.